### PR TITLE
Secure Cookies

### PR DESCRIPTION
**Pull request by jmera**
See #145 for initial discussion

We would like to give users the ability to flag all cookies as secure: https://www.owasp.org/index.php/SecureFlag

### CODE REVIEW

**Comment by oreoshake** [comment]
I'll drop 1.9.3 in the next major release, but for now please make travis happy.

File: lib/secure_headers/middleware.rb [hunk 1]
```diff
@@ -8,8 +10,32 @@ def initialize(app)
...
+            "#{cookie}; secure"
+          else
+            cookie
+          end
+        end.join("\n")
+      end
+    end
+
+    def config(req)
+      req.env[SECURE_HEADERS_CONFIG] || Configuration.get(Configuration::DEFAULT_CONFIG)
```
Comment (jmera) [inline lib/secure_headers/middleware.rb]: @oreoshake is there an easier way to access the config from this context?

Comment (oreoshake) [inline lib/secure_headers/middleware.rb]: `req.env[SECURE_HEADERS_CONFIG]` should suffice.

File: lib/secure_headers/railtie.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Railtie < Rails::Railtie
                              'Public-Key-Pins', 'Public-Key-Pins-Report-Only']
 
       initializer "secure_headers.middleware" do
-        Rails.application.config.middleware.use SecureHeaders::Middleware
+        Rails.application.config.middleware.insert_before 0, SecureHeaders::Middleware
```
Comment (jmera) [inline lib/secure_headers/railtie.rb]: For this feature to work correctly `SecureHeaders::Middleware` needs to come before the session middleware (or simply at the top of the rack stack).

File: lib/secure_headers/middleware.rb (hunk 1 above)
Comment (oreoshake) [inline lib/secure_headers/middleware.rb]: Or maybe make `config_for` public? https://github.com/twitter/secureheaders/blob/44c8c3edfe6c5456274c448ba68371b20fe649d7/lib/secure_headers.rb#L225

File: lib/secure_headers/railtie.rb (hunk 2 above)
Comment (oreoshake) [inline lib/secure_headers/railtie.rb]: This should be fine.

File: lib/secure_headers/middleware.rb (hunk 1 above)
Comment (jmera) [inline lib/secure_headers/middleware.rb]: Sounds good to me, will help keep things dry

**Comment by oreoshake** [comment]
Woot. I'll release this with https://github.com/twitter/secureheaders/pull/232 tomorrow.

**Comment by jmera** [comment]
:beers:

**Pull request merged**